composer/trainer/trainer.py

```diff
@@ -592,31 +762,37 @@ def _ddp_reduce_tensor_sum(tensor: Tensor) -> Tensor:
 
                     # if resuming, skip dataloader forward to the minibatch index
                     if batch_idx < self.state.batch_idx:
-                        if self.checkpoint_loader:
+                        if self.checkpoint_loader is not None:
                             self.checkpoint_loader.restore_checkpoint_rng_state(self.state, self.device)
+                            self.checkpoint_loader = None
                         continue
 
                     state.last_batch_size = self._get_batch_size(state.batch)
                     state.batch = self.device.batch_to_device(state.batch)
                     if self._train_device_transformation_fn is not None:
                         state.batch = self._train_device_transformation_fn(state.batch)
 
+                    if self._train_split_fn is None:
+                        split_fn = default_batch_split_fn
+                    else:
+                        split_fn = self._train_split_fn
```

Comment:
```suggestion
                    split_fn = default_batch_split_fn if self._train_split_fn is None else self._train_split_fn
```

or.. put this logic closer to where the `self._train_split_fn`  is defined?